aiidalab_eln/cheminfo/__init__.py

```diff
@@ -72,5 +98,31 @@ def is_connected(self):
     @traitlets.default("eln_type")
     def set_eln_type(self):
         return "cheminfo"
-    
-    
+
+    def set_sample_config(self, **kwargs):
+        if 'sample_uuid' in kwargs:
+            self.sample_uuid = kwargs['sample_uuid']
+        if 'file_name' in kwargs:
+            self.file_name = kwargs['file_name']
+        if 'spectrum_type' in kwargs:
+            self.spectrum_type = kwargs['spectrum_type']
+        
+    def sample_config_editor(self):
+        return ipw.VBox(
+            [
+                self.sample_uuid_widget,
+                self.file_name_widget,
+                self.spectrum_type_widget,
+            ]
+        )
+
+    def send_data_object(self, data_object):
+        sample_manager = self.session.get_sample(self.sample_uuid)
+
+        if data_object.node_type == "data.dict.Dict.":
+            from .exporter import export_isotherm
+            export_isotherm(sample_manager, isotherm=data_object, adsorptive="N2", filename=self.file_name)
```

Comment:
do we want to leave N2 hard coded?

Reply:
And likely we should catch some errors in this exporter to show in the UI. For example, this AuthenticationError we discussed.

Reply:
I think that whole thing should/will be replaced with an isotherm data type where adsorptive will be stored as well

Reply:
> For example, this AuthenticationError we discussed.

This I am checking in the aiidalab-widgets-base (https://github.com/aiidalab/aiidalab-widgets-base/pull/181).

Reply:
> And likely we should catch some errors in this exporter to show in the UI

At the same time, we should catch the "no rights" error and raise a unified exception, so that the GUI understands that.

Reply:
>At the same time, we should catch the "no rights" error and raise a unified exception, so that the GUI understands that.

this is what I meant ;) we'll raise https://github.com/cheminfo-py/cheminfopy/blob/4324d79049c9e7f83bc8e53f73b1d405f621154d/src/cheminfopy/errors.py#L10 in this cheminfopy package